src/main/flight/failsafe.c

```diff
@@ -63,27 +64,39 @@ failsafeState_t* failsafeInit(rxConfig_t *intialRxConfig)
     rxConfig = intialRxConfig;
 
     failsafeState.events = 0;
-    failsafeState.enabled = false;
+    failsafeState.monitoring = false;
 
     return &failsafeState;
 }
 
-bool failsafeIsIdle(void)
+failsafePhase_e failsafePhase()
 {
-    return failsafeState.counter == 0;
+    return failsafeState.phase;
 }
 
-bool failsafeIsEnabled(void)
+#define MAX_COUNTER_VALUE_WHEN_RX_IS_RECEIVED_AFTER_RX_CYCLE 1
```

Comment:
This issue will be solved if the counter is replaced with timestamps, see comments in #781 regarding RX_SERIAL and RX_MSP.